src/EditorFeatures/Core/Implementation/KeywordHighlighting/HighlightingService.cs

```diff
@@ -27,11 +28,21 @@ internal class HighlightingService : IHighlightingService
         public IEnumerable<TextSpan> GetHighlights(
              SyntaxNode root, int position, CancellationToken cancellationToken)
         {
-            return _highlighters.Where(h => h.Metadata.Language == root.Language)
-                               .Select(h => h.Value.GetHighlights(root, position, cancellationToken))
-                               .WhereNotNull()
-                               .Flatten()
-                               .Distinct();
+            try
+            {
+                return _highlighters.Where(h => h.Metadata.Language == root.Language)
```

Comment:
this one was crashing because root was null. root is null since csharp document was under xaml document which has no syntaxtree so, we got null root here. and throw